docs/source/api/roles.rst

```diff
@@ -65,19 +84,236 @@ Response Structure
+
+``PUT``
+=======
+.. versionadded:: 1.3
+
+Replaces an existing :term:`Role` with one provided by the request.
+
+:Auth. Required: Yes
+:Roles Required: "admin"
+:Response Type:
+
+Request Structure
+-----------------
+.. table:: Request Query Parameters
+
+	+------+----------+--------------------------------------------------------------------+
+	| Name | Required | Description                                                        |
+	+======+==========+====================================================================+
+	| id   | yes      | The integral, unique identifier of the :term:`Role` to be replaced |
+	+------+----------+--------------------------------------------------------------------+
+
+:capabilities: An optional array of capability names that will be granted to the new :term:`Role`
+
+	.. warning:: When not present, the affected :term:`Role`'s Capabilities will be unchanged - *not* removed, unlike when the array is empty.
+
+:description: A helpful description of the :term:`Role`'s purpose.
+:name:        The new name of the :term:`Role`
+:privLevel:   The new privilege level of the new :term:`Role`\ [#privlevel]_
+
+.. code-block:: http
+	:caption: Request Example
+
+	PUT /api/1.3/roles?id=5 HTTP/1.1
+	Host: trafficops.infra.ciab.test
+	User-Agent: curl/7.47.0
+	Accept: */*
+	Cookie: mojolicious=...
+	Content-Length: 56
+	Content-Type: application/json
+
+	{
+		"name":"test",
+		"privLevel": 29,
+		"description": "quest"
+	}
+
+Response Structure
+------------------
+:capabilities: An array of the names of the Capabilities given to this :term:`Role`
+
+	.. tip:: This can be ``null`` *or* empty, depending on whether it was present in the request body, or merely empty. Obviously, it can also be a populated array.
+
+	.. warning:: If no ``capabilities`` array was given in the request, this will *always* be ``null``, even if the :term:`Role` has Capabilities that would have gone unchanged.
+
+:description: A description of the :term:`Role`
+:id:          The integral, unique identifier for this :term:`Role`
+:name:        The name of the :term:`Role`
+:privLevel:   An integer that allows for comparison between :term:`Roles`
+
+.. code-block:: http
+	:caption: Response Example
+
+	HTTP/1.1 200 OK
+	Access-Control-Allow-Credentials: true
+	Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept, Set-Cookie, Cookie
+	Access-Control-Allow-Methods: POST,GET,OPTIONS,PUT,DELETE
+	Access-Control-Allow-Origin: *
+	Content-Type: application/json
+	Set-Cookie: mojolicious=...; Path=/; HttpOnly
+	Whole-Content-Sha512: mlHQenE1Q3gjrIK2lC2hfueQOaTCpdYEfboN0A9vYPUIwTiaF5ZaAMPQBdfGyiAhgHRxowITs3bR7s1L++oFTQ==
+	X-Server-Name: traffic_ops_golang/
+	Date: Thu, 05 Sep 2019 12:56:46 GMT
+	Content-Length: 150
+
+	{
+		"alerts": [
+			{
+				"text": "role was updated.",
+				"level": "success"
+			}
+		],
+		"response": {
+			"id": 5,
+			"name": "test",
+			"description": "quest",
+			"privLevel": 29,
+			"capabilities": null
```

Comment:
do we return null responses to our users instead of empty throughout our API? I usually dont like doing that as then client have to code against it but if we use it many other places then this is fine

Reply:
No idea. Generally I think an empty list would be better, but this documentation is accurate to how the endpoint behaves (which I don't *believe* I changed in this PR) and that's all I was attempting to accomplish.

Reply:
fine to leave - might be good to do a sweep through all our endpoint for 2.0 and cleanup any null responses

Reply:
The 2.0 todo list is long